### PR TITLE
Slims down disposal unit ui and fixes thrown collisions

### LINKED ISSUE #3875: Throwing items in disposal bin can appear broken
## Description
When throwing an item in a disposal bin, theres a 25% of failing. In my opionion, this can just feel broken to a player (see #3768 😳). I think we should either remove the random chance, or make a small message pop up that says "Missed!" or something.

### PR DESCRIPTION
**Pull request by SweptWasTaken**
## About the PR 

- Cleaned up disposal unit ui
- Localized disposal unit ui
- Fixed throwing objects into disposal unit
- Organized the disposal/atmos pipe yaml

After looking at the mess that is disposal mailing unit ui and realizing it's broken in game too i've commented out the yaml for it as it's busted currently.

Closes #3875.
 

**Screenshots**

UI before
image

UI after
image

**Changelog**

:cl:
- fix: Throwing entities into disposal units now work
- tweak: Slims down disposal unit UI
- remove: Disposal mailing units for now

### CODE REVIEW

File: Content.Client/VendingMachines/VendingMachineMenu.cs [hunk 1]
```diff
@@ -60,7 +61,8 @@ public void Populate(List<VendingMachineInventoryEntry> inventory)
                 _items.AddItem($"{itemName} ({entry.Amount} left)", icon);
             }
 
-            SetSize = ((longestEntry.Length + 8) * 12, _items.Count * 40 + 50);
+            SetSize = (Math.Clamp((longestEntry.Length + 8) * 12, 300, 400),
```
Comment (mirrorcult) [inline Content.Client/VendingMachines/VendingMachineMenu.cs:64]: ?

Comment (SweptWasTaken) [reply Content.Client/VendingMachines/VendingMachineMenu.cs:64]: Idiot moment, was testing your PR kek.

**Review by ShadowCommander — CHANGES_REQUESTED** [review]

File: Content.Client/GameObjects/Components/Disposal/DisposalUnitWindow.cs [hunk 2]
```diff
@@ -16,15 +18,14 @@ public class DisposalUnitWindow : SS14Window
...
         public readonly Button Eject;
         public readonly Button Power;
 
         public DisposalUnitWindow()
         {
-            MinSize = SetSize = (300, 225);
-
+            IoCManager.InjectDependencies(this);
+            MinSize = SetSize = (300, 140);
             Contents.AddChild(new VBoxContainer
```
[the remark this answers is not in the document]
Comment (SweptWasTaken) [reply Content.Client/GameObjects/Components/Disposal/DisposalUnitWindow.cs:29]: Sure but i'll do it in another PR.

File: Content.Client/GameObjects/Components/Disposal/DisposalUnitWindow.cs [hunk 3]
```diff
@@ -16,15 +18,14 @@ public class DisposalUnitWindow : SS14Window
...
         public readonly Button Engage;
         public readonly Button Eject;
         public readonly Button Power;
 
         public DisposalUnitWindow()
         {
-            MinSize = SetSize = (300, 225);
-
+            IoCManager.InjectDependencies(this);
+            MinSize = SetSize = (300, 140);
```
Comment (ShadowCommander) [inline Content.Client/GameObjects/Components/Disposal/DisposalUnitWindow.cs:28]: Set Resizable to false maybe?

File: Content.Client/GameObjects/Components/Disposal/DisposalUnitWindow.cs [hunk 4]
```diff
@@ -33,28 +34,26 @@ public DisposalUnitWindow()
                     {
                         Children =
                         {
-                            new Label {Text = Loc.GetString("State: ")},
-                            (_unitState = new Label {Text = Loc.GetString("Ready")})
+                            new Label {Text = Loc.GetString("ui-disposal-unit-label-state")},
+                            new Control {MinSize = (4, 0)},
```
Comment (ShadowCommander) [inline Content.Client/GameObjects/Components/Disposal/DisposalUnitWindow.cs:38]: Can probably replace this with `HBoxContainer.SeparationOverride = 4` on the above HBoxContainer.

File: Content.Client/GameObjects/Components/Disposal/DisposalUnitWindow.cs [hunk 5]
```diff
@@ -33,28 +34,26 @@ public DisposalUnitWindow()
...
                     },
-                    new Control {MinSize = (0, 10)},
+                    new Control {MinSize = (0, 5)},
                     new HBoxContainer
                     {
                         Children =
                         {
-                            new Label {Text = Loc.GetString("Pressure:")},
+                            new Label {Text = Loc.GetString("ui-disposal-unit-label-pressure")},
+                            new Control {MinSize = (4, 0)},
```
Comment (ShadowCommander) [inline Content.Client/GameObjects/Components/Disposal/DisposalUnitWindow.cs:48]: Same as above.

File: Resources/Prototypes/Entities/Constructible/Piping/Disposal/units.yml [hunk 6]
```diff
@@ -0,0 +1,99 @@
+- type: entity
+  abstract: true
+  id: DisposalUnitBase
+  parent: BaseConstructible
```
Comment (ShadowCommander) [inline Resources/Prototypes/Entities/Constructible/Piping/Disposal/units.yml:4]: Can you use BaseMachinePowered or is this too different?

**Review by ShadowCommander — CHANGES_REQUESTED** [review]

**Comment by SweptWasTaken** [comment]
Reviews applied.

**Review by ShadowCommander — APPROVED** [review]

**Pull request merged**